pkg/server/test/check.go

```diff
@@ -146,15 +391,105 @@ func BenchmarkCheck(b *testing.B, ds storage.OpenFGADatastore) {
 		b.Run(name, func(b *testing.B) {
 			for i := 0; i < b.N; i++ {
 				response, _, err := checkQuery.Execute(ctx, &openfgav1.CheckRequest{
-					StoreId:          storeID,
-					TupleKey:         bm.tupleKey,
-					ContextualTuples: bm.contextualTuples,
-					Context:          bm.contextStruct,
+					StoreId:  storeID,
+					TupleKey: bm.tupleKeyToCheck,
+					Context:  bm.contextGenerator(),
 				})
 
 				require.Equal(b, bm.expected, response.GetAllowed())
 				require.NoError(b, err)
 			}
 		})
 	}
+
+	// Now run the one benchmark that doesn't fit the table pattern
+	benchmarkCheckWithBypassUsersetReads(b, ds)
+}
+
+// This benchmark test creates multiple authorization models so it doesn't fit into
+// the table pattern above.
+func benchmarkCheckWithBypassUsersetReads(b *testing.B, ds storage.OpenFGADatastore) {
+	schemaOne := `
+		model
+			schema 1.1
+		type user
+		type group
+			relations
+				define member: [user]
+		type document
+			relations
+				define viewer: [user:*, group#member]
+	`
+	storeID := ulid.Make().String()
+	modelOne := testutils.MustTransformDSLToProtoWithID(schemaOne)
+
+	err := ds.WriteAuthorizationModel(context.Background(), storeID, modelOne)
+	require.NoError(b, err)
+
+	// add user to many usersets according to model A
+	var tuples []*openfgav1.TupleKey
+	for i := 0; i < 1000; i++ {
+		tuples = append(tuples, &openfgav1.TupleKey{
+			Object:   fmt.Sprintf("group:%d", i),
+			Relation: "member",
+			User:     "user:anne",
+		})
+	}
+
+	// one userset gets access to document:budget
+	tuples = append(tuples, &openfgav1.TupleKey{Object: "document:budget", Relation: "viewer", User: "group:999#member"})
+
+	// now actually write the tuples
+	for i := 0; i < len(tuples); {
+		var tuplesToWrite []*openfgav1.TupleKey
+		for j := 0; j < ds.MaxTuplesPerWrite(); j++ {
+			if i == len(tuples) {
+				break
+			}
+			tuplesToWrite = append(tuplesToWrite, tuples[i])
+			i++
+		}
+		err := ds.Write(context.Background(), storeID, nil, tuplesToWrite)
+		require.NoError(b, err)
+	}
+
+	schemaTwo := `
+		model
+			schema 1.1
+		type user
+		type user2
+		type group
+			relations
+				define member: [user2]
+		type document
+			relations
+				define viewer: [user:*, group#member]
+	`
+	modelTwo := testutils.MustTransformDSLToProtoWithID(schemaTwo)
+	typeSystemTwo, err := typesystem.NewAndValidate(context.Background(), modelTwo)
+	require.NoError(b, err)
+
+	// all the usersets added above are now invalid and should be skipped!
+	err = ds.WriteAuthorizationModel(context.Background(), storeID, modelTwo)
+	require.NoError(b, err)
+
+	checkQuery := commands.NewCheckCommand(
+		ds,
+		graph.NewLocalChecker(),
+		typeSystemTwo,
```

Comment:
I _think_ the second typesystem is accomplishing the same goal as the original test scenario but someone please double check me